fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LucenePartitioner.java

```diff
@@ -589,15 +557,53 @@ private CompletableFuture<LucenePartitionInfoProto.LucenePartitionInfo> getOrCre
     }
 
     /**
-     * get the partition metadata containing the given partitioning key or fail if not found.
+     * try to get the partition to which a given record belongs.
      *
-     * @param partitioningKey partitioning key
-     * @return partition metadata future
-     * @throws RecordCoreException if no partition is found
+     * @param record record
+     * @param groupingKey grouping key
+     * @param <M> message
+     * @return null future if no suitable partition exists, partition info otherwise
      */
     @Nonnull
-    private CompletableFuture<LucenePartitionInfoProto.LucenePartitionInfo> getPartitionInfoOrFail(@Nonnull Tuple groupingKey, @Nonnull Tuple partitioningKey) {
-        return assignPartitionInternal(groupingKey, partitioningKey, false);
+    <M extends Message> CompletableFuture<LucenePartitionInfoProto.LucenePartitionInfo> tryGetPartitionInfo(
+            @Nonnull FDBIndexableRecord<M> record,
+            @Nonnull Tuple groupingKey) {
+        if (!isPartitioningEnabled()) {
+            return CompletableFuture.completedFuture(null);
+        }
+        return assignPartitionInternal(groupingKey, toPartitionKey(record), false)
```

Comment:
Instead of catching an exception here, `assignPartitionInternal` should take an `enum` for whether it should create, or we should have different methods composed so that this returns `null`.